editor/script_create_dialog.cpp

```diff
@@ -163,21 +163,28 @@ bool ScriptCreateDialog::_can_be_built_in() {
 	return (supports_built_in && built_in_enabled);
 }
 
+String ScriptCreateDialog::_adjust_file_path(const String &p_base_path) const {
+	if (p_base_path.is_empty()) {
+		return p_base_path;
+	}
+
+	ScriptLanguage *current_script_language = ScriptServer::get_language(current_language);
+	String prefix = p_base_path.get_base_dir();
+	String file_name = p_base_path.get_file().get_basename();
+	file_name = EditorNode::adjust_script_name_casing(file_name, current_script_language->preferred_file_name_casing());
+	String suffix = current_script_language->get_extension();
```

Comment:
```suggestion
	String extension = current_script_language->get_extension();
```
The names could be clearer. `prefix` can be `base_path` or `base_dir`.